src/tests/test_sexual_behaviour.py

```diff
@@ -104,3 +106,225 @@ def test_initial_sex_behaviour_groups():
             Expectation = float(n_sex)*p
             N_g = sum(index_sex & index_group)
             assert Expectation - sigma*3 <= N_g <= Expectation + sigma*3
+
+
+def test_rred_long_term_partner():
+    N = 1000
+    pop = Population(size=N, start_date=date(1989, 1, 1))
+    # pick some indices and give those people LTPs
+    indices = [randint(0, N-1) for i in range(15)]
+    pop.data["partnered"] = False
+    pop.data.loc[indices, "partnered"] = True
+    SBM = SexualBehaviourModule()
+    SBM.update_rred_long_term_partnered(pop.data)
+    for i in indices:
+        assert pop.data.loc[i, "rred_long_term_partnered"] == SBM.ltp_risk_factor
+
+
+def test_rred_adc():
+    N = 20
+    pop = Population(size=N, start_date=date(1989, 1, 1))
+    pop.data["HIV_status"] = False  # make test independent of how HIV is initialised
+    init_HIV_idx = [randint(0, N-1) for i in range(5)]
+    pop.data.loc[init_HIV_idx, "HIV_status"] = True
+    expected_rred = np.ones(N)
+    expected_rred[init_HIV_idx] = 0.2
+    SBM = SexualBehaviourModule()
+    # initialise rred_adc
+    SBM.init_risk_factors(pop.data)
+    # Check only people with HIV have rred_adc = 0.2
+    assert np.all(pop.data["rred_adc"] == expected_rred)
+    # Assign more people with HIV
+    add_HIV_idx = [randint(0, N-1) for i in range(5)]
+    pop.data.loc[add_HIV_idx, "HIV_status"] = True
+    # Update rred factors
+    SBM.update_sex_behaviour(pop)
+    expected_rred[add_HIV_idx] = 0.2
+    assert np.all(pop.data["rred_adc"] == expected_rred)
+
+
+def test_rred_diagnosis():
+    N = 20
+    pop = Population(size=N, start_date=date(1989, 1, 1))
+    pop.data["HIV_status"] = False  # init everyone HIV negative
+    SBM = SexualBehaviourModule()
+    SBM.rred_diagnosis = 4
+    SBM.rred_diagnosis_period = timedelta(700)
+    SBM.update_rred_diagnosis(pop.data, pop.date)
+    assert np.all(pop.data["rred_diagnosis"] == 1)
+    # give up some people HIV and advance the date
+    HIV_idx = [randint(0, N-1) for i in range(5)]
+    pop.data.loc[HIV_idx, "HIV_status"] = True
+    pop.data.loc[HIV_idx, "HIV_Diagnosis_Date"] = pop.date
+    SBM.update_rred_diagnosis(pop.data, pop.date)
+    for i in HIV_idx:
+        assert pop.data.loc[i, "rred_diagnosis"] == 4
+    pop.date += timedelta(days=365)
+    SBM.update_rred_diagnosis(pop.data, pop.date)
+    for i in HIV_idx:
+        assert pop.data.loc[i, "rred_diagnosis"] == 4
+    pop.date += timedelta(days=500)
+    SBM.update_rred_diagnosis(pop.data, pop.date)
+    for i in HIV_idx:
+        assert pop.data.loc[i, "rred_diagnosis"] == 2
```

Comment:
Can also check that people outside `HIV_idx` still have the old value? (either at the end or after each step)